toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,270 @@
-							connectedToWebApp = true;
-							NotificationManager.showMessage("Web Agent - Connected to Webapp !").showNotification();
-						}
-						else
-							NotificationManager.showMessage("The ApiKey does not match with the WebApp: \n" + webConfigProvider.get().getApiKey()).showNotification();
-					}
-					else 
-						NotificationManager.showMessage("The Web App does not anwser").showNotification();
-				} catch (IOException e1) {
-					LOG.error(e1.getMessage(), e1);
-				}
+	        	connect();
 	        }
 	    };
 	    return listener;
 	}
 	
-	ActionListener getKillListener(){
+	//TODO: Switch implementation to websocket and have a connection watcher
+	private void connect() {
+		try {
+			if(isWebAppListening()) {
+				if(agentServer.register(webConfigProvider.get().getApiKey())) {
+					makeAgentOnline();
+					NotificationManager.showMessage("Web Agent - Connected to Webapp !").showNotification();
+				}
+				else{
+					NotificationManager.showMessage("The ApiKey does not match with the WebApp: \n" + webConfigProvider.get().getApiKey()).showNotification();
+				}
+			}
+			else {
+				NotificationManager.showMessage("The Web App does not anwser").showNotification();
+			}
+		} catch (IOException e1) {
+			LOG.error(e1.getMessage(), e1);
+		}
+	}
+
+	private void makeAgentOnline() {
+		trayIcon.setImage(onlineImage);
+		connectedToWebApp = true;
+	}
+
+	private boolean isWebAppListening() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_TEST_WEB_APP_URL);
+	}
+
+	private ActionListener getExecuteListener(){
 	    ActionListener listener = new ActionListener() {
 	        public void actionPerformed(ActionEvent e) {
-	        	agentServer.unRegister();
-	        	System.exit(-1);
+	        	execute();
+	        }
+	    };
+	    return listener;
+	}
+	
+	private void execute() {
+		try {
+			if(hasPluginDirectory() && hasScriptsDirectory()) {
+				NotificationManager.showMessage(MainAppMessages.SCRIPTS_EXECUTION).showNotification();
+				Thread thread = new Thread(new WaiterThread(webConfigProvider));
+				thread.start();
+			}
+		} catch (IOException e1) {
+			LOG.error(e1.getMessage(), e1);
+		}
+	}
+
+	private boolean hasPluginDirectory() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_PLUGIN_DIR);
+	}
+
+	private boolean hasScriptsDirectory() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_SCRIPTS_DIR);
+	}
+
+	private ActionListener getKillListener(){
+	    ActionListener listener = new ActionListener() {
+	        public void actionPerformed(ActionEvent event) {
+				try {
+					agentServer.unRegister();
+				} catch (UnknownHostException exception) {
+					LOG.error(exception.getMessage(), exception);
+				}finally {
+					System.exit(0);
+				}
+
 	        }
 	    };
 	    return listener;
 	}
 	
-	ActionListener getStartListener(){
+	private ActionListener getStartListener(){
 	    ActionListener listener = new ActionListener() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1604)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "listener". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)